### PR TITLE
[TM-140] View Application Details

### PR DESCRIPTION
**Pull request by mmiddleton3301**
Summary:

* Exposes out `IsNamePublic` on the parent pledge when getting an individual application;
* Exposes out the pledge id that the application relates to

See also:

* https://github.com/SkillsFundingAgency/das-apim-endpoints/pull/497
* https://github.com/SkillsFundingAgency/das-levy-transfer-matching-web/pull/92

### CODE REVIEW

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]
See outer api tech review - this ticket requires a new endpoint.

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/GetApplication/GetApplicationResult.cs [hunk 1]
```diff
@@ -21,11 +20,13 @@ public class GetApplicationResult
...
         public string EmployerAccountName { get; set; }
-        public List<PledgeLocation> PledgeLocations { get; set; }
+        public List<string> PledgeLocations { get; set; }
         public ApplicationStatus Status { get; set; }
         public IEnumerable<Sector> PledgeSectors { get; set; }
         public IEnumerable<Level> PledgeLevels { get; set; }
         public IEnumerable<JobRole> PledgeJobRoles { get; set; }
         public int PledgeRemainingAmount { get; set; }
+        public bool PledgeIsNamePublic { get; set; }
+        public int PledgeId { get; set; }
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching/Application/Queries/GetApplication/GetApplicationResult.cs:30]: Could you also add the Id of the sender and receiver, as we will need this here shortly.

Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching/Application/Queries/GetApplication/GetApplicationResult.cs:30]: Done.

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/GetApplication/GetApplicationQueryHandler.cs [hunk 2]
```diff
@@ -23,8 +23,17 @@ public async Task<GetApplicationResult> Handle(GetApplicationQuery request, Canc
                 .Include(x => x.EmailAddresses)
                 .Include(x => x.EmployerAccount)
                 .Include(x => x.Pledge)
+                .Include(x => x.Pledge.EmployerAccount)
                 .Include(x => x.Pledge.Locations)
-                .SingleOrDefaultAsync(x => x.Id == request.Id, cancellationToken);
+                .Where(
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching/Application/Queries/GetApplication/GetApplicationQueryHandler.cs:28]: I don't find this particularly readable personally. See `GetApplicationsQueryHandler.cs` for an example where a query object is created and parameters conditionally applied to it prior to projection.

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Api/Controllers/ApplicationsController.cs [hunk 3]
```diff
@@ -31,7 +31,32 @@ public async Task<IActionResult> GetApplication(int pledgeId, int applicationId)
...
             {
-                Id = applicationId,
+                PledgeId = pledgeId,
+                ApplicationId = applicationId,
+            });
+
+            IActionResult result = null;
+            if (queryResult != null)
+            {
+                result = new OkObjectResult((GetApplicationResponse)queryResult);
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Api/Controllers/ApplicationsController.cs:41]: Good point. Was having a mad moment I think. Fixed.

File: src/SFA.DAS.LevyTransferMatching.Api/Controllers/ApplicationsController.cs [hunk 4]
```diff
@@ -31,7 +31,32 @@ public async Task<IActionResult> GetApplication(int pledgeId, int applicationId)
         {
             var queryResult = await _mediator.Send(new GetApplicationQuery()
             {
-                Id = applicationId,
+                PledgeId = pledgeId,
+                ApplicationId = applicationId,
+            });
+
+            IActionResult result = null;
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Api/Controllers/ApplicationsController.cs:38]: Point taken - changed.

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/GetApplication/GetApplicationQueryHandler.cs (hunk 2 above)
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching/Application/Queries/GetApplication/GetApplicationQueryHandler.cs:28]: Done.

**Pull request merged**